py/client/pydeephaven/session.py

```diff
@@ -239,24 +209,39 @@ def exportable_objects(self) -> Dict[str, ticket_pb2.TypedTicket]:
             fields = self._fetch_fields()
             return {field.field_name: field.typed_ticket for field in fields if field.application_id == 'scope'}
 
+    def update_metadata(self, metadata: Iterable[Tuple[bytes, bytes]]):
+        for header_tuple in metadata:
+            if header_tuple[0] == "authorization":
+                new_auth_value = header_tuple[1]
+                with self._auth_lock:
+                    self._auth_value = bytes(new_auth_value, 'ascii')
+                break
+
     @property
     def grpc_metadata(self):
-        l = [(b'authorization', self._auth_token)]
+        with self._auth_lock:
+            if self._auth_value is not None and isinstance(self._auth_value, bytes) and len(self._auth_value) > 0:
+                auth_value = deepcopy_bytes(self._auth_value)
+                l = [(b'authorization', auth_value)]
+            else:
+                l = []
         if self._extra_headers:
             l.extend(list(self._extra_headers.items()))
         return l
 
     @property
     def table_service(self) -> TableService:
-        if not self._table_service:
-            self._table_service = TableService(self)
-        return self._table_service
+        with self._r_lock:
+            if not self._table_service:
+                self._table_service = TableService(self)
+            return self._table_service
```

Comment:
I don't see a reason to change what we were doing before, which was to try to avoid initializing what we don't need.  I just changed it to be consistent and consistently use the lock.

Do you have any specific concerns of issues that could result from either the change here in this PR or the general lazy initialization approach that is already being taken by the current code (but unlocked so wrong in the current code in the sense of being read/modify/write sequences without locking)